popup.js

```
@@ -269,6 +269,149 @@ function hideLimitWarnings() {
 
+// ============================================================================
+// TOKEN LIMIT & CONTENT TRIMMING UTILITIES
+// ============================================================================
+
+/**
+ * Estimate token count from character count
+ * Uses approximate ratio of 4 characters per token
+ * @param {number} charCount - Number of characters
+ * @returns {number} - Estimated token count
+ */
+function estimateTokens(charCount) {
```

Biome (2.4.4)

[error] Shouldn't redeclare 'estimateTokens'. Consider to delete it or rename it.

(lint/suspicious/noRedeclare)